### PR TITLE
[ABW-3096] Security center

### PR DESCRIPTION
**Pull request by kugel3**
Jira ticket: ABW-3096

## Description
Security Center, with Configuration backup and integrated with Security Factors

## Video
Coming

## PR submission checklist
- [x] I have tested account to account transfer flow and have confirmed that it works

### CODE REVIEW

File: RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (CyonAlexRDX) [inline RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: maybe should revert bumping ?

File: Gemfile.lock
Comment (CyonAlexRDX) [inline Gemfile.lock:1]: maybe should revert bumping ?

File: RadixWallet/Core/FeaturePrelude/RadixDateFormatter.swift
Comment (CyonAlexRDX) [inline RadixWallet/Core/FeaturePrelude/RadixDateFormatter.swift:1]: We ought to move this to Sargon post 1.6 release

File: RadixWallet/Prelude/Extensions/IdentifiedArrayOf+Extensions.swift [hunk 1]
```diff
@@ -6,6 +6,22 @@ extension IdentifiedArrayOf {
 	}
 }
 
+// MARK: - ForEachStatic
+/// A wrapper around ForEach that can be used with **static** collections of elements that don't conform to `Identifiable`
+public struct ForEachStatic<Data: RandomAccessCollection, Content: View>: View {
```
Comment (CyonAlexRDX) [inline RadixWallet/Prelude/Extensions/IdentifiedArrayOf+Extensions.swift:11]: I'd prefer it if we did not use `Data` as generic type name. Maybe `Elements` ? which will read out:
```swift
public let elements: [OffsetIdentified<Elements.Element>]
```

**Review by CyonAlexRDX — APPROVED** [review]
Missing those pretty video recordings :) nothing major just left som minor remarks!

File: RadixWallet/Clients/CloudBackupClient/CloudBackupClient+Interface.swift [hunk 2]
```diff
@@ -36,12 +39,13 @@ public struct CloudBackupClient: DependencyKey, Sendable {
...
 	public typealias LoadDeviceID = @Sendable () async -> UUID?
 	public typealias MigrateProfilesFromKeychain = @Sendable () async throws -> [CKRecord]
-	public typealias DeleteProfileBackup = @Sendable (UUID) async throws -> Void
+	public typealias DeleteProfileBackup = @Sendable (ProfileID) async throws -> Void
 	public typealias CheckAccountStatus = @Sendable () async throws -> CKAccountStatus
-	public typealias LastBackup = @Sendable (UUID) -> AnyAsyncSequence<BackupResult?>
-	public typealias LoadProfile = @Sendable (UUID) async throws -> Profile?
+	public typealias LastBackup = @Sendable (ProfileID) -> AnyAsyncSequence<BackupResult?>
+	public typealias LoadProfile = @Sendable (ProfileID) async throws -> Profile?
 	public typealias LoadAllProfiles = @Sendable () async throws -> [Profile]
```
Comment (danvleju-rdx) [inline RadixWallet/Clients/CloudBackupClient/CloudBackupClient+Interface.swift:49]: We should return `containsP2PLinks` for every profile (i.e. `[(Profile, Bool)]`).
Using profile data [here](https://github.com/radixdlt/babylon-wallet-ios/pull/1122/files#diff-c4ae62b3f1ebe8cd50673c681cd202bc3b22760a37e90016f7781f9fd94acdc8R60) you can make this check:
```swift
Profile.checkIfProfileJsonContainsLegacyP2PLinks(contents: data)
```

File: RadixWallet/Features/ProfileBackupsFeature/RestoreProfileFromBackup/Children/SelectBackup/SelectBackup+Reducer.swift [hunk 3]
```diff
@@ -111,22 +116,13 @@ public struct SelectBackup: Sendable, FeatureReducer {
...
-					return
-				}
-				await send(.delegate(.selectedProfileSnapshot(snapshot, isInCloud: true, containsP2PLinks: containsP2PLinks)))
-			} catch: { error, send in
-				loggerGlobal.error("Failed to load profile snapshot with header, error: \(error), header: \(profileHeader)")
-				await send(.internal(.snapshotWithHeaderNotFoundInCloud(profileHeader)))
-			}
+		case let .tappedUseCloudBackup(profile):
+			// TODO: pass p2pLinks
+			return .send(.delegate(.selectedProfile(profile, isInCloud: true, containsP2PLinks: false)))
```
Comment (danvleju-rdx) [inline RadixWallet/Features/ProfileBackupsFeature/RestoreProfileFromBackup/Children/SelectBackup/SelectBackup+Reducer.swift:125]: So if `cloudBackupClient.loadAllProfiles()` returns `containsP2PLinks` for every profile, this action will also pass the `containsP2PLinks` flag.

**Review by danvleju-rdx — CHANGES_REQUESTED** [review]
Looks good! Only few changes needed to adapt to p2p links migration.

**Review by GhenadieVP — APPROVED** [review]

**Pull request merged**